src/Agent.Plugins/PipelineArtifact/Telemetry/FileShareActionRecord.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using Agent.Sdk;
+using Microsoft.VisualStudio.Services.Content.Common.Telemetry;
+
+namespace Agent.Plugins.PipelineArtifact.Telemetry
+{
+    /// <summary>
+    /// Generic telemetry record for use with FileShare Artifact events.
+    /// </summary>
+    public class FileShareActionRecord : PipelineArtifactActionRecord
+    {
+        public int TotalFile { get; private set; }
+        public long TotalContentSize { get; private set; }
+        public long TimeLapse { get; private set; }
+        public string Command { get; private set; }
+        public int ExitCode { get; private set; }
+        public IList<ArtifactRecord> ArtifactRecords { get; private set; }
+
+        public FileShareActionRecord(TelemetryInformationLevel level, Uri baseAddress, string eventNamePrefix, string eventNameSuffix, AgentTaskPluginExecutionContext context, uint attemptNumber = 1)
+            : base(level, baseAddress, eventNamePrefix, eventNameSuffix, context, attemptNumber)
+        {
+        }
+
+        protected override void SetMeasuredActionResult<T>(T value)
+        {
+            if (value is FileSharePublishResult)
+            {
+                FileSharePublishResult result = value as FileSharePublishResult;
+                Command = result.Command;
+                ExitCode = result.ExitCode;
+            }
+
+            if (value is FileShareDownloadResult)
+            {
+                FileShareDownloadResult result = value as FileShareDownloadResult;
+                TotalFile = result.FileCount;
+                TotalContentSize = result.ContentSize;
+                this.ArtifactRecords = result.ArtifactRecords;
+            }
+        }
+    }
+ 
+    public sealed class FileSharePublishResult
+    {
+        public string Command { get; private set; }
+        public int ExitCode { get; private set; }
+
+        public FileSharePublishResult(string command, int exitCode)
+        {
+            this.Command = command;
+            this.ExitCode = exitCode;
+        }
+    }
+
+    public sealed class FileShareDownloadResult
+    {
+        public int FileCount { get; private set; }
+        public long ContentSize { get; private set; }
+        public IList<ArtifactRecord> ArtifactRecords { get; private set; }
+
+        public FileShareDownloadResult(IList<ArtifactRecord> records, int fileCount, long contentSize)
+        {
+            this.FileCount = fileCount;
+            this.ContentSize = contentSize;
+            this.ArtifactRecords = records;
+        }
+    }
+
+    public sealed class ArtifactRecord
+    {
+        public string ArtifactName { get; private set; }
+        public int FileCount { get; private set; }
+        public long ContentSize { get; private set; }
+        public long TimeLapse { get; private set; }
```

Comment:
>TimeLapse [](start = 20, length = 9)

TimeLapse is what unit (seconds, milliseconds)? Append the unit to the end of the variable name